modules/dists/BlockDist.chpl

```diff
@@ -1291,7 +1304,8 @@ proc BlockArr.dsiRankChange(d, param newRank: int, param stridable: bool, args)
 
 proc BlockArr.dsiReindex(d: BlockDom) {
   var alias = new BlockArr(eltType=eltType, rank=d.rank, idxType=d.idxType,
-                           stridable=d.stridable, dom=d);
+                           stridable=d.stridable, dom=d,
+                           sparseLayoutType=DefaultDist);
```

Comment:
This line seems to be an example of passing `sparseLayoutType` as an argument and not having a default gets ugly.

dsiReindex (and very similarly in dsiRankChange and dsiSlice) is a method on BlockArr that has nothing to do with SparseBlockArr, yet has to have `sparseLayoutType` as a generic argument simply because of hooks to BlockDom.